File: Java/4.17.java
/*
Write a program whose input is a character and a string, and whose output indicates the number of times the character appears in the string.

Ex: If the input is:

n Monday
the output is:

1
Ex: If the input is:

z Today is Monday
the output is:

0
Ex: If the input is:

n It's a sunny day
the output is:

2
*/
import java.util.Scanner;

public class LabProgram {
   public static void main(String[] args) {
      /* Type your code here. */
      Scanner scan = new Scanner(System.in);
      
      char inputChar = scan.next().charAt(0);
      String inputString = scan.nextLine();
      int count = 0;
      
      for (int i = 0; i < inputString.length() - 1; i++) {
         if (inputChar == inputString.charAt(i)) {
            count++;
         }
      }
      System.out.printf("%d\n", count);
   }
}

File: CPP/4.17.cpp
/*
Write a program whose input is a character and a string, and whose output indicates the number of times the character appears in the string. The output should include the input character and use the plural form, n's, if the number of times the characters appears is not exactly 1.

Ex: If the input is:

n Monday
the output is:

1 n
Ex: If the input is:

z Today is Monday
the output is:

0 z's
Ex: If the input is:

n It's a sunny day
the output is:

2 n's
Case matters.

Ex: If the input is:

n Nobody
the output is:

0 n's
n is different than N.
*/

#include <iostream>
#include <string>
using namespace std;

int main() {

   /* Type your code here. */
   char userInputLetter;
   string userInputString;
   unsigned int count = 0;
   
   cin >> userInputLetter;
   getline(cin, userInputString);
   
   for (int i = 0; i < userInputString.length(); i++) {
      if (userInputLetter == userInputString[i]) {
         count++;
      }
   }
   
   if (count == 1) {
      printf("%d %c\n", count, userInputLetter);
   }
   else {
      printf("%d %c's\n", count, userInputLetter);
   }

   return 0;
}
